tools/glide-pindeps/pindeps.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/openshift/origin/tools/glide-pindeps/glide"
+)
+
+func readFile(filepath string) ([]byte, error) {
+	f, err := os.Open(filepath)
+	if err != nil {
+		return nil, err
+	}
+
+	defer f.Close()
+
+	b, err := ioutil.ReadAll(f)
+	if err != nil {
+		return nil, err
+	}
+
+	return b, nil
+}
+
+func readLockFile(filepath string) (*glide.LockFile, error) {
+	b, err := readFile(filepath)
+	if err != nil {
+		return nil, err
+	}
+
+	glidelock := &glide.LockFile{}
+	glidelock.Decode(b)
+
+	return glidelock, err
+}
+
+func readYamlFile(filepath string) (*glide.YamlFile, error) {
+	b, err := readFile(filepath)
+	if err != nil {
+		return nil, err
+	}
+
+	glideyaml := &glide.YamlFile{}
+	glideyaml.Decode(b)
+
+	return glideyaml, err
+}
+
+func main() {
+	var lockFilePath string
+	var glideFilePath string
+
+	flag.Usage = func() {
+		usageText := `This program reads a dependency tree from a glide.lock file and adds or updates dependencies on a glide.yaml file that are either missing or outdated.
+Any existing dependencies on a glide.yaml file containing a "repo" field are ignored.
+`
+		fmt.Fprintf(os.Stderr, "%s\n", usageText)
+		flag.PrintDefaults()
+	}
+	flag.StringVar(&lockFilePath, "lockfile", "glide.lock", "Path to glide.lock file used to indicate which dependencies should be updated or pinned to a specific revision in the glide.yaml file")
+	flag.StringVar(&glideFilePath, "yamlfile", "glide.yaml", "Path to glide.yaml file used to pin dependencies from the given lockfile")
+	flag.Parse()
+
+	if len(lockFilePath) == 0 {
```

Comment:
Call a function for all the rest of this.  If the function returns an error, exit(1).